test/dlc_tests/sanity/test_telemetry.py

```diff
@@ -0,0 +1,139 @@
+    job_type = test_utils.get_job_type_from_image(image_uri)
+    processor = test_utils.get_processor_from_image_uri(image_uri)
+
+    container_name = f"{repo_name}-telemetry_bad_instance_role-ec2"
+
+    docker_cmd = "nvidia-docker" if processor == "gpu" else "docker"
+
+    test_utils.login_to_ecr_registry(ec2_connection, account_id, image_region)
+    ec2_connection.run(f"{docker_cmd} pull -q {image_uri}")
+
+    preexisting_ec2_instance_tags = ec2_utils.get_ec2_instance_tags(ec2_instance_id, ec2_client=ec2_client)
+    if expected_tag_key in preexisting_ec2_instance_tags:
+        ec2_client.remove_tags(Resources=[ec2_instance_id], Tags=[{"Key": expected_tag_key}])
+
+    # Disable access to EC2 instance metadata
+    ec2_connection.run(f"sudo route add -host 169.254.169.254 reject")
+
+    if framework == "tensorflow" and job_type == "inference":
+        env_vars_list = ecs_utils.get_ecs_tensorflow_environment_variables(processor, "saved_model_half_plus_two")
+        env_vars = " ".join([f"-e {entry['name']}={entry['value']}" for entry in env_vars_list])
+        ec2_connection.run(f"{docker_cmd} run {env_vars} --name {container_name} -id {image_uri}")
+        time.sleep(5)
+    else:
+        ec2_connection.run(f"{docker_cmd} run --name {container_name} -id {image_uri} bash")
+        output = ec2_connection.run(
+            f"{docker_cmd} exec -i {container_name} python -c 'import {framework}; import time; time.sleep(5)'",
+            warn=True
+        )
+        assert output.ok, f"'import {framework}' fails when credentials not configured"
+
+    ec2_instance_tags = ec2_utils.get_ec2_instance_tags(ec2_instance_id, ec2_client=ec2_client)
+    assert expected_tag_key not in ec2_instance_tags, (
+        f"{expected_tag_key} was applied as an instance tag."
+        "EC2 create_tags went through even though it should not have"
+    )
+
+
+def _run_tag_success(image_uri, ec2_client, ec2_instance, ec2_connection):
+    expected_tag_key = "aws-dlc-autogenerated-tag-do-not-delete"
+
+    ec2_instance_id, _ = ec2_instance
+    account_id = test_utils.get_account_id_from_image_uri(image_uri)
+    image_region = test_utils.get_region_from_image_uri(image_uri)
+    repo_name, image_tag = test_utils.get_repository_and_tag_from_image_uri(image_uri)
+    framework, _ = test_utils.get_framework_and_version_from_tag(image_uri)
+    job_type = test_utils.get_job_type_from_image(image_uri)
+    processor = test_utils.get_processor_from_image_uri(image_uri)
+
+    container_name = f"{repo_name}-telemetry_tag_instance_success-ec2"
+
+    docker_cmd = "nvidia-docker" if processor == "gpu" else "docker"
+
+    test_utils.login_to_ecr_registry(ec2_connection, account_id, image_region)
+    ec2_connection.run(f"{docker_cmd} pull -q {image_uri}")
+
+    preexisting_ec2_instance_tags = ec2_utils.get_ec2_instance_tags(ec2_instance_id, ec2_client=ec2_client)
+    if expected_tag_key in preexisting_ec2_instance_tags:
```

Comment:
why explicitly delete existing tag?

Reply:
The validation for the test is that a tag has been applied correctly. Therefore, by applying a tag to the instance (unintentionally or not), it would be possible to make the test succeed even when the `deep_learning_container.py` script has actually failed to apply a tag on the instance.

Reply:
ok. I guess example could be like:

1. Someone manually created instance tag.
2. Some framework being import has been done before this?

Reply:
Exactly.

Also, we create separate instances for each test, right now. But it is possible, in theory, that we would refactor our tests to reduce this parallelism (maybe to cut down on the number of instances created?). This bit of code will ensure that a previously-run test on the same instance doesn't corrupt the preconditions for this test.

Reply:
cool